packages/client/src/index.ts

```diff
@@ -0,0 +1,189 @@
+import axios, { AxiosError, AxiosPromise } from "axios";
+import * as t from "io-ts";
+
+import { UserProfile } from "@slicemachine/core/build/models";
+import { SharedSlice } from "@prismicio/types-internal/lib/customtypes/widgets/slices";
+import { CustomType } from "@prismicio/types-internal/lib/customtypes/CustomType";
+
+import type { ClientError, UploadParameters } from "./models";
+import {
+  ApplicationMode,
+  getStatus,
+  getMessage,
+  ApisEndpoints,
+  ProductionApisEndpoints,
+  StageApisEndpoints,
+  AclCreateResult,
+} from "./models";
+
+import { getAndValidateResponse } from "./utils";
+import { upload } from "./utils/upload";
+
+// exporting models to be used with the Client.
+export * from "./models";
+
+// exporting utils so the client calls can be replicated for speficic calls to a consumer.
+export * from "./utils";
+
+// exporting the client itself.
+export class Client {
+  apisEndpoints: ApisEndpoints;
+  repository: string | null;
+  authenticationToken: string;
+
+  constructor(
+    applicationMode: ApplicationMode,
+    repository: string | null,
+    authenticationToken: string
+  ) {
+    this.repository = repository;
+    this.authenticationToken = authenticationToken;
+
+    if (applicationMode === ApplicationMode.PROD)
+      this.apisEndpoints = ProductionApisEndpoints;
+    else if (applicationMode === ApplicationMode.STAGE)
+      this.apisEndpoints = StageApisEndpoints;
+    else {
+      // Dev
+      this.apisEndpoints = ProductionApisEndpoints;
+    }
+  }
+
+  // setters to provide flexibility
+  updateAuthenticationToken(newToken: string) {
+    this.authenticationToken = newToken;
+  }
+
+  updateRepository(repository: string | null) {
+    this.repository = repository;
+  }
+
+  // private methods
+  _fetch(
+    method: "get" | "post",
+    url: string,
+    data?: Record<string, unknown>,
+    headers?: Record<string, unknown>
+  ): AxiosPromise {
```

Comment:
No type paramters?

Reply:
What do you mean ?
Like an object and typing the whole object ?